### ISSUE TITLE
UnusedImports False Positive in Android when importing View extension function/property

### ISSUE THREAD

**Issue by mhernand40** [detekt/detekt#4029]
Take the following code snippet from a sample Android project:
```Kotlin
package com.detekt.issues

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.isVisible
import com.detekt.lib.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.textView.isVisible = false
    }
}

```

## Expected Behavior
Since all imports are used, Detekt should not complain about `UnusedImports`.

## Observed Behavior
When running `detektMain`, it complains with the following error:
```
style - 5min debt
        UnusedImports - [MainActivity.kt] at *****/src/main/java/com/detekt/issues/MainActivity.kt:5:1
```

## Steps to Reproduce
1. Please unzip the following sample project.
[DetektUnusedImportsFalsePostivie.zip](https://github.com/detekt/detekt/files/6964494/DetektUnusedImportsFalsePostivie.zip)
2. From the terminal, navigate into the project's root directory.
3. Run `./gradlew detektMain`
4. Observe Detekt failure regarding `UnusedImports`.

## Context
In Android projects, it is quite common to define extension functions for `Views`. Android KTX provides many useful extension functions out of the box. With false positives such as these, unfortunately this `UnusedImports` rule cannot be enabled for Android projects.

## Your Environment

* Version of detekt used: `1.17.1`
* Version of Gradle used (if applicable):
```
------------------------------------------------------------
Gradle 7.0.2
------------------------------------------------------------

Build time:   2021-05-14 12:02:31 UTC
Revision:     1ef1b260d39daacbf9357f9d8594a8a743e2152e

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.9 (Eclipse OpenJ9 openj9-0.23.0)
OS:           Mac OS X 10.16 x86_64
```
* Gradle scan link (add `--scan` option when running the gradle task): https://gradle.com/s/d3slx4ce74r5q
* Operating System and version: MacOS Big Sur Version 11.5
* Link to your project (if it's a public repository): N/A (please see attached zip file)

**Comment by mhernand40** [comment]
Verified that this issue still exists in version `1.18.0`.

**Comment by cortinico** [comment]
Thanks for the report @mhernand40 and for the reproducer.
Are you able to work on a fix for this false positive? We can provide some guidance if you get stuck somewhere :)

**Comment by mhernand40** [comment]
You're welcome @cortinico. If it's not urgent then I can probably take it in a few weeks or so. I have a few major things going on for at least the next 2-3 weeks. Because I only have a handful of contributions in this repo, it always takes me a bit to ramp up and get familiar with the APIs again. 😅

**Comment by BraisGabin** [comment]
> Because I only have a handful of contributions in this repo, it always takes me a bit to ramp up and get familiar with the APIs again. 😅

That's a very valuable feedback! If you find something that we could do to make this easier please open an issue! Or just pointing out what is the difficult part. A high entry barrier is an issue that I would like to remove (or at least reduce).

**Comment by severn-everett** [comment]
I'm taking a look at this but cannot make a reproduction using a pure-Kotlin POC. One thing I noticed when running the reproduction steps listed above was this output:
```
//Users/marcelohernandez/AndroidStudioProjects/DetektUnusedImportsFalsePostivie/lib/src/main/java/com/detekt/issues/MainActivity.kt:6:23: error: unresolved reference: databinding
import com.detekt.lib.databinding.ActivityMainBinding
                      ^
//Users/marcelohernandez/AndroidStudioProjects/DetektUnusedImportsFalsePostivie/lib/src/main/java/com/detekt/issues/MainActivity.kt:11:23: error: unresolved reference: ActivityMainBinding
        val binding = ActivityMainBinding.inflate(layoutInflater)
                      ^
//Users/marcelohernandez/AndroidStudioProjects/DetektUnusedImportsFalsePostivie/lib/src/main/java/com/detekt/issues/MainActivity.kt:13:26: error: variable expected
        binding.textView.isVisible = false
                         ^
```
This leads me to believe that there might be a linking issue where Detekt can't conduct the reference lookup correctly. However, I'm unsure of how I'd reproduce this with a home-grown POC in the Android context - my experience with Kotlin is purely on the JVM side, not with any Android development.

**Comment by BraisGabin** [comment]
I see... I think that we can't fix it now then. #4465 will fix this.

As a workaround you can use our first-party plugin `formatting` or `ktlint` directly. Both have another implementation of this rule that should work.

I'm going to close this because the fix is being thacked in other issue and there is a workaround to this meanwhile. Thanks @severn-everett to take a lot at it :)

**Comment by joeboyscout04** [comment]
Being a detekt newbie, I wonder if you can explain a bit on how to make the workaround work:

>As a workaround you can use our first-party plugin formatting or ktlint directly. Both have another implementation of this rule that should work.

is this something you enable in build.gradle separately from `detekt`? Does it take it's own `ruleset`?

**Comment by cortinico** [comment]
> is this something you enable in build.gradle separately from `detekt`? Does it take it's own `ruleset`?

It's explained here: https://detekt.github.io/detekt/index.html#adding-more-rule-sets
You need to add a `detektPlugins` dependency on `detekt-formatting` and then you can use the `NoUnusedImport` rule that has a similar behavior to `UnusedImports` https://detekt.github.io/detekt/formatting.html#nounusedimports
